src/sentry/api/endpoints/project_release_commits.py

```diff
@@ -54,24 +54,32 @@ def get(self, request: Request, project, version) -> Response:
         repo_name = request.query_params.get("repo_name")
 
         # prefer repo external ID to name
+        # NOTE: We filter on Repository here instead of using get b/c sometimes,
+        # we may have multiple repos with the same external_id/name but diff
+        # provider names due to plugins vs integrations.
+        # (e.g "bitbucket" for the plugin vs "integrations:bitbucket" for the integration).
         if repo_id:
-            try:
-                repo = Repository.objects.get(
-                    organization_id=organization_id, external_id=repo_id, status=ObjectStatus.ACTIVE
-                )
-                queryset = queryset.filter(commit__repository_id=repo.id)
-            except Repository.DoesNotExist:
+            repos = Repository.objects.filter(
+                organization_id=organization_id, external_id=repo_id, status=ObjectStatus.ACTIVE
+            ).order_by("-date_added")
+
+            if not repos.exists():
                 raise ResourceDoesNotExist
 
-        elif repo_name:
-            try:
-                repo = Repository.objects.get(
-                    organization_id=organization_id, name=repo_name, status=ObjectStatus.ACTIVE
-                )
-                queryset = queryset.filter(commit__repository_id=repo.id)
-            except Repository.DoesNotExist:
+            latest_repo = repos.first()
```

Comment:
nit: you can also just do

```
latest_repo = repos.first()
if latest_repo is None:
    raise ResourceDoesNotExist
```

which saves a query because `repos.exists()` takes a query to evaluate

Reply:
nice catch